app/src/main/java/com/example/movieapp/presentation/extensions/FragmentExt.kt

```diff
@@ -1,10 +1,19 @@
 package com.example.movieapp.presentation.extensions
 
+import android.app.Dialog
+import android.content.Context
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.LiveData
+import com.example.movieapp.databinding.DialogErrorBinding
 import com.google.android.material.snackbar.Snackbar
 
 fun Fragment.createSnackBar(message: String, color: Int) {
     Snackbar.make(requireView(), message, Snackbar.LENGTH_LONG).apply {
         setTextColor(color)
     }.show()
-}
\ No newline at end of file
+}
+
+fun <T> Fragment.observeData(liveData: LiveData<T>, observer: (it: T) -> Unit) {
+    liveData.observe(this.viewLifecycleOwner, { observer(it) })
+}
+
```

Comment:
fun <T> LiveData<T>.observe(owner: LifecycleOwner, f: (T?) -> Unit) {
    this.observe(owner, object : Observer<T> {
        override fun onChanged(t: T?) {
            f(t)
        }

    })
}

შეგიძლია ესეც გააკეთო, ლაივდატას ექსთენშენი და არა ფრაგმენტის

Reply:
აქ ფრაგმენტის გადაწოდება მაინც გვიწევს , ზურამ ოვნერების გადაწოდების გარე გააკეთეთო